projects/auth0-angular/src/lib/auth.client.ts

```diff
@@ -1,14 +1,22 @@
 import { InjectionToken } from '@angular/core';
 import { Auth0Client } from '@auth0/auth0-spa-js';
-import { AuthConfig } from './auth.config';
+import { AuthClientConfig } from './auth.config';
 import useragent from '../useragent';
 
 export class Auth0ClientFactory {
-  static createClient(config: AuthConfig): Auth0Client {
+  static createClient(configFactory: AuthClientConfig): Auth0Client {
+    const config = configFactory.get();
+
+    if (!config) {
+      throw new Error(
+        'Configuration must be specified either through AuthModule.forRoot or through AuthClientConfig.set'
+      );
+    }
+
     const { redirectUri, clientId, maxAge, ...rest } = config;
 
     return new Auth0Client({
-      redirect_uri: redirectUri,
+      redirect_uri: redirectUri || window.location.origin,
```

Comment:
If the Angular platform is a server then `window` will be undefined.

For example `@angular/universal` would throw during SSR.

Use of `isPlatformBrowser` feels like an antipattern but may be appropriate.
```
// DI Parameter
@Inject(PLATFORM_ID) protected platformId

// Platform Check
if (isPlatformBrowser(this.platformId)) {
    redirectUri = window.location.origin;
}
```
**Reference**

https://github.com/angular/universal/issues/830
https://github.com/angular/universal/issues/1523
https://stackoverflow.com/questions/60188777/angular-9-universal-referenceerror-window-is-not-defined

Reply:
This is a good call. I would argue to discuss (and tackle) this as part of a separate PR as `window.location.origin` is already used. This PR is just moving the usage.

Reply:
SSR support is something we'll be exploring in the future based on customer feedback. The underlying SPA SDK that this Angular SDK uses does present some problems with doing that, but there are changes coming to that too which may pave the way for SSR support.

As @frederikprijck, the change in the PR doesn't make it worse in that regard, it's just shifting logic around.